metric/config.go

```diff
@@ -20,109 +20,117 @@ import (
+	instrumentationName    string
+	instrumentationVersion string
+}
+
+// Description describes the instrument in human-readable terms.
+func (cfg InstrumentConfig) Description() string {
+	return cfg.description
+}
+
+// Unit describes the measurement unit for a instrument.
+func (cfg InstrumentConfig) Unit() unit.Unit {
+	return cfg.unit
+}
+
+// InstrumentationName is the name of the library providing
+// instrumentation.
+func (cfg InstrumentConfig) InstrumentationName() string {
+	return cfg.instrumentationName
+}
+
+// InstrumentationVersion is the version of the library providing
+// instrumentation.
+func (cfg InstrumentConfig) InstrumentationVersion() string {
+	return cfg.instrumentationVersion
 }
 
 // InstrumentOption is an interface for applying metric instrument options.
 type InstrumentOption interface {
 	// ApplyMeter is used to set a InstrumentOption value of a
 	// InstrumentConfig.
-	ApplyInstrument(*InstrumentConfig)
+	applyInstrument(*InstrumentConfig)
 }
 
 // NewInstrumentConfig creates a new InstrumentConfig
 // and applies all the given options.
 func NewInstrumentConfig(opts ...InstrumentOption) InstrumentConfig {
 	var config InstrumentConfig
 	for _, o := range opts {
-		o.ApplyInstrument(&config)
+		o.applyInstrument(&config)
 	}
 	return config
 }
 
-// WithDescription applies provided description.
-func WithDescription(desc string) InstrumentOption {
-	return descriptionOption(desc)
-}
+type instrumentOptionFunc func(*InstrumentConfig)
 
-type descriptionOption string
+func (fn instrumentOptionFunc) applyInstrument(cfg *InstrumentConfig) {
+	fn(cfg)
+}
 
-func (d descriptionOption) ApplyInstrument(config *InstrumentConfig) {
-	config.Description = string(d)
+// WithDescription applies provided description.
+func WithDescription(desc string) InstrumentOption {
+	return instrumentOptionFunc(func(cfg *InstrumentConfig) {
+		cfg.description = desc
+	})
 }
 
 // WithUnit applies provided unit.
 func WithUnit(unit unit.Unit) InstrumentOption {
-	return unitOption(unit)
-}
-
-type unitOption unit.Unit
-
-func (u unitOption) ApplyInstrument(config *InstrumentConfig) {
-	config.Unit = unit.Unit(u)
+	return instrumentOptionFunc(func(cfg *InstrumentConfig) {
+		cfg.unit = unit
+	})
 }
 
 // WithInstrumentationName sets the instrumentation name.
 func WithInstrumentationName(name string) InstrumentOption {
-	return instrumentationNameOption(name)
-}
-
-type instrumentationNameOption string
-
-func (i instrumentationNameOption) ApplyInstrument(config *InstrumentConfig) {
-	config.InstrumentationName = string(i)
+	return instrumentOptionFunc(func(cfg *InstrumentConfig) {
+		cfg.instrumentationName = name
+	})
 }
 
 // MeterConfig contains options for Meters.
 type MeterConfig struct {
 	// InstrumentationVersion is the version of the library providing
 	// instrumentation.
-	InstrumentationVersion string
+	instrumentationVersion string
+}
+
+func (cfg MeterConfig) InstrumentationVersion() string {
```

Comment:
Missing comment on exported method.

```suggestion
// InstrumentationVersion returns the version of the library providing instrumentation.
func (cfg MeterConfig) InstrumentationVersion() string {
```